### ISSUE TITLE
Tests do not build

### ISSUE THREAD

**Issue by mhier** [ChimeraTK/ControlSystemAdapter-OPC-UA-Adapter#19]
When enabling the tests (cmake options " -DENABLE_UNIT_TESTS=1 -DENABLE_COVERAGE=1"), I get the following compiler errors (on Ubuntu 16.04 / gcc 5.4.0):

```
Scanning dependencies of target ControlSystem-OPCUA_Sample_Adapter
[ 57%] Building CXX object CMakeFiles/ControlSystem-OPCUA_Sample_Adapter.dir/examples/ControlSystem_OPCUA_Sample_Adapter.cpp.o
[ 60%] Linking CXX executable ControlSystem-OPCUA_Sample_Adapter
[ 60%] Built target ControlSystem-OPCUA_Sample_Adapter
Scanning dependencies of target test_opcua_processvariable
[ 64%] Building CXX object tests/CMakeFiles/test_opcua_processvariable.dir/src/test_opcua_processvariable.cpp.o
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/tests/src/test_opcua_processvariable.cpp: In static member function ‘static void ProcessVariableTest::testEmptySet()’:
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/tests/src/test_opcua_processvariable.cpp:52:190: error: no matching function for call to ‘mtca_processvariable::mtca_processvariable(UA_Server*&, UA_NodeId&, const string&, const string&, const char [1], const char [1], boost::shared_ptr<ChimeraTK::ControlSystemPVManager>&)’
   mtca_processvariable *test = new mtca_processvariable(serverSet->mappedServer, serverSet->baseNodeId, oneProcessVariable->getName(), oneProcessVariable->getName(), "", "", pvSet.csManager);
                                                                                                                                                                                              ^
In file included from /home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_uaadapter.h:34:0,
                 from /home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/tests/src/test_opcua_processvariable.cpp:1:
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:51:2: note: candidate: mtca_processvariable::mtca_processvariable(UA_Server*, UA_NodeId, std::__cxx11::string, std::__cxx11::string, std::__cxx11::string, std::__cxx11::string, boost::shared_ptr<ChimeraTK::ControlSystemPVManager>, boost::shared_ptr<ChimeraTK::ControlSystemSynchronizationUtility>)
  mtca_processvariable(UA_Server *server, UA_NodeId basenodeid, string namePV, string nameNew, string engineeringUnit, string description, boost::shared_ptr<ControlSystemPVManager> csManager, boost::shared_ptr<Co
  ^
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:51:2: note:   candidate expects 8 arguments, 7 provided
In file included from /home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_uaadapter.h:34:0,
                 from /home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/tests/src/test_opcua_processvariable.cpp:1:
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:50:2: note: candidate: mtca_processvariable::mtca_processvariable(UA_Server*, UA_NodeId, std::__cxx11::string, boost::shared_ptr<ChimeraTK::ControlSystemPVManager>, boost::shared_ptr<ChimeraTK::ControlSystemSynchronizationUtility>)
  mtca_processvariable(UA_Server *server, UA_NodeId basenodeid, string namePV, boost::shared_ptr<ControlSystemPVManager> csManager, boost::shared_ptr<ControlSystemSynchronizationUtility> syncCsUtility);
  ^
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:50:2: note:   candidate expects 5 arguments, 7 provided
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:37:7: note: candidate: mtca_processvariable::mtca_processvariable(const mtca_processvariable&)
 class mtca_processvariable : ua_mapped_class {
       ^
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:37:7: note:   candidate expects 1 argument, 7 provided
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/tests/src/test_opcua_processvariable.cpp: In static member function ‘static void ProcessVariableTest::testClientSide()’:
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/tests/src/test_opcua_processvariable.cpp:189:140: error: no matching function for call to ‘mtca_processvariable::mtca_processvariable(UA_Server*&, UA_NodeId&, const string&, boost::shared_ptr<ChimeraTK::ControlSystemPVManager>&)’
   varList.push_back(new mtca_processvariable(serverSet->mappedServer, serverSet->baseNodeId, oneProcessVariable->getName(), pvSet.csManager)); 
                                                                                                                                            ^
In file included from /home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_uaadapter.h:34:0,
                 from /home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/tests/src/test_opcua_processvariable.cpp:1:
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:51:2: note: candidate: mtca_processvariable::mtca_processvariable(UA_Server*, UA_NodeId, std::__cxx11::string, std::__cxx11::string, std::__cxx11::string, std::__cxx11::string, boost::shared_ptr<ChimeraTK::ControlSystemPVManager>, boost::shared_ptr<ChimeraTK::ControlSystemSynchronizationUtility>)
  mtca_processvariable(UA_Server *server, UA_NodeId basenodeid, string namePV, string nameNew, string engineeringUnit, string description, boost::shared_ptr<ControlSystemPVManager> csManager, boost::shared_ptr<Co
  ^
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:51:2: note:   candidate expects 8 arguments, 4 provided
In file included from /home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_uaadapter.h:34:0,
                 from /home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/tests/src/test_opcua_processvariable.cpp:1:
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:50:2: note: candidate: mtca_processvariable::mtca_processvariable(UA_Server*, UA_NodeId, std::__cxx11::string, boost::shared_ptr<ChimeraTK::ControlSystemPVManager>, boost::shared_ptr<ChimeraTK::ControlSystemSynchronizationUtility>)
  mtca_processvariable(UA_Server *server, UA_NodeId basenodeid, string namePV, boost::shared_ptr<ControlSystemPVManager> csManager, boost::shared_ptr<ControlSystemSynchronizationUtility> syncCsUtility);
  ^
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:50:2: note:   candidate expects 5 arguments, 4 provided
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:37:7: note: candidate: mtca_processvariable::mtca_processvariable(const mtca_processvariable&)
 class mtca_processvariable : ua_mapped_class {
       ^
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/include/mtca_processvariable.h:37:7: note:   candidate expects 1 argument, 4 provided
tests/CMakeFiles/test_opcua_processvariable.dir/build.make:62: recipe for target 'tests/CMakeFiles/test_opcua_processvariable.dir/src/test_opcua_processvariable.cpp.o' failed
make[2]: *** [tests/CMakeFiles/test_opcua_processvariable.dir/src/test_opcua_processvariable.cpp.o] Error 1
CMakeFiles/Makefile2:393: recipe for target 'tests/CMakeFiles/test_opcua_processvariable.dir/all' failed
make[1]: *** [tests/CMakeFiles/test_opcua_processvariable.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by mhier** [comment]
Sorry, it's quite bad to read. The first error message is this:


```
/home/mhier/ChimeraTK/sources/ControlSystemAdapter-OPC-UA-Adapter/tests/src/test_opcua_processvariable.cpp:52:190: error: no matching function for call to ‘mtca_processvariable::mtca_processvariable(UA_Server*&, UA_NodeId&, const string&, const string&, const char [1], const char [1], boost::shared_ptr<ChimeraTK::ControlSystemPVManager>&)’
   mtca_processvariable *test = new mtca_processvariable(serverSet->mappedServer, serverSet->baseNodeId, oneProcessVariable->getName(), oneProcessVariable->getName(), "", "", pvSet.csManager);

```

**Comment by mhier** [comment]
Danke dir! Jetzt failen die Tests bei mir allerdings, gibt's da noch ein bekanntes Problem? 

> From: "Julian Rahm" <notifications@github.com>
> To: "ChimeraTK/ControlSystemAdapter-OPC-UA-Adapter"
> <ControlSystemAdapter-OPC-UA-Adapter@noreply.github.com>
> Cc: "Martin Hierholzer" <martin.hierholzer@desy.de>, "Author"
> <author@noreply.github.com>
> Sent: Thursday, 8 December, 2016 12:51:32
> Subject: Re: [ChimeraTK/ControlSystemAdapter-OPC-UA-Adapter] Tests do not build
> (#19)

> Closed #19 via 9232321 .

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub , or mute the thread .

-- 
Martin Hierholzer 
DESY 
Notkestrasse 85 
22607 Hamburg 
Office 55a / 123 
Tel: +49 40 8998 1897 
Mob: +49 176 992 46 476 
Skype: mhier48 

This e-mail may contain confidential and/or privileged information and is intended only for the use of the individual or entity named above. If you received it in error, please advise the sender immediately by reply e-mail and delete the original. Any further use - especially any form of publication - of this e-mail without explicit authorisation is strictly prohibited.
